src/backend/tests/routers/test_agent.py

```diff
@@ -1,11 +1,50 @@
+from unittest.mock import patch
+
+import pytest
 from fastapi.testclient import TestClient
 from sqlalchemy.orm import Session
 
 from backend.config.deployments import ModelDeploymentName
 from backend.config.tools import ToolName
 from backend.database_models.agent import Agent
 from backend.database_models.agent_tool_metadata import AgentToolMetadata
+from backend.services.metrics import report_metrics
 from backend.tests.factories import get_factory
+from backend.schemas.metrics import (
+    MetricsData,
+    MetricsMessageType,
+)
+
+
+@pytest.mark.asyncio
+async def test_create_agent_mertic(
+    session_client: TestClient, session: Session
+) -> None:
+    user = get_factory("User", session).create(fullname="John Doe")
+    request_json = {
+        "name": "test agent",
+        "version": 1,
+        "description": "test description",
+        "preamble": "test preamble",
+        "temperature": 0.5,
+        "model": "command-r-plus",
+        "deployment": ModelDeploymentName.CoherePlatform,
+        "tools": [ToolName.Calculator, ToolName.Search_File, ToolName.Read_File],
+    }
+
+    with patch(
+        "backend.services.metrics.report_metrics",
+        return_value=None,
+    ) as mock_metrics:
+        response = session_client.post(
+            "/v1/agents", json=request_json, headers={"User-Id": user.id}
+        )
+        assert response.status_code == 200
+        m_args: MetricsData = mock_metrics.await_args.args[0]
+        assert m_args.user_id == user.id
+        assert m_args.message_type == MetricsMessageType.ASSISTANT_CREATED
+        assert m_args.assistant is not None
+        assert m_args.user.fullname == user.fullname
 
```

Comment:
nice!

should we consider creating a new test directory for metrics(or middlewares in general)? i feel like this file is going to get crowded if we also add the metric tests (same with all the other router test files). also would be nice to separate concerns. wdyt?

Reply:
aligned offline to co locate metrics and router tests since they are currently so coupled